conda_lock/conda_lock.py

```diff
@@ -59,6 +62,38 @@
 CONDA_PKGS_DIRS = None
 DEFAULT_PLATFORMS = ["osx-64", "linux-64", "win-64"]
 
+_PLATFORM_MAP = {"osx-64": "darwin", "linux-64": "linux", "win-64": "win"}
+
+
+def _extract_platform(line):
+    search = PLATFORM_PATTERN.search(line)
+    if search:
+        return search.group(1)
+
+
+def extract_platform(lockfile: str) -> str:
+    for line in lockfile.strip().split("\n"):
+        platform = _extract_platform(line)
+        if platform:
+            return platform
+    raise RuntimeError("Cannot find platform in lockfile.")
+
+
+def _do_validate_platform(platform: str) -> Tuple[bool, str]:
+    return sys.platform.startswith(_PLATFORM_MAP[platform]), sys.platform
+
+
+def do_validate_platform(lockfile: str):
+    platform_lockfile = extract_platform(lockfile)
+    try:
+        success, platform_sys = _do_validate_platform(platform_lockfile)
+    except KeyError:
+        raise RuntimeError(f"Unknown platform type in lockfile '{platform_lockfile}'.")
+    if not success:
+        raise RuntimeError(
+            f"Platform in lockfile '{platform_lockfile}' is not compatible with system platform '{platform_sys}'."
+        )
```

Comment:
I think it would be better to raise a custom `ValidationError`, especially since I catch that error type laster. But I didn't want to go overboard and introduce custom error classes before hearing what you think about that.

Reply:
Adding some custom error classes is fine by me